src/AutoRest.CSharp.V3/Generation/Writers/ClientWriter.cs

```diff
@@ -331,6 +342,145 @@ private void WritePagingOperation(CodeWriter writer, Paging pagingMethod, bool a
+                    {
+                        case JsonSerialization jsonSerialization:
+                            writer.Append($"using var {document:D} = ");
+                            writer.Line($"{typeof(JsonDocument)}.Parse(response.ContentStream);");
+                            writer.ToDeserializeCall(
+                                jsonSerialization,
+                                w => w.Append($"document.RootElement"),
+                                ref valueVariable
+                            );
+                            writer.Line($"return {typeof(Response)}.FromValue({valueVariable}, response);");
+                            break;
+                        case XmlElementSerialization xmlSerialization:
+                            writer.Line($"var {document:D} = {typeof(XDocument)}.Load(response.ContentStream, LoadOptions.PreserveWhitespace);");
+                            writer.ToDeserializeCall(
+                                xmlSerialization,
+                                w => w.Append($"document"),
+                                ref valueVariable
+                            );
+                            writer.Line($"return {typeof(Response)}.FromValue({valueVariable}, response);");
+                            break;
+                        default:
+                            writer.Line($"return Response.FromValue(response, response);");
+                            break;
+                    }
+                }
+
+                using (writer.Scope($"async (response, cancellationToken) =>", "{", "});"))
+                {
+                    switch (serialization)
+                    {
+                        case JsonSerialization jsonSerialization:
+                            writer.Append($"using var {document:D} = ");
+                            writer.Line($"await {typeof(JsonDocument)}.ParseAsync(response.ContentStream, default, cancellationToken).ConfigureAwait(false);");
+                            writer.ToDeserializeCall(
+                                jsonSerialization,
+                                w => w.Append($"document.RootElement"),
+                                ref valueVariable
+                            );
+                            writer.Line($"return {typeof(Response)}.FromValue({valueVariable}, response);");
+                            break;
+                        case XmlElementSerialization xmlSerialization:
+                            writer.Line($"var {document:D} = {typeof(XDocument)}.Load(response.ContentStream, LoadOptions.PreserveWhitespace);");
+                            writer.ToDeserializeCall(
+                                xmlSerialization,
+                                w => w.Append($"document"),
+                                ref valueVariable
+                            );
+                            writer.Line($"return {typeof(Response)}.FromValue({valueVariable}, response);");
```

Comment:
Let's get rid of `Response<T>` creation too.

Reply:
We use both the Response itself and the Value for the UpdateStatus:
![image](https://user-images.githubusercontent.com/17788297/75382589-9d3fe780-588f-11ea-81e0-9eff189fce61.png)

Do you want to return something less elegant, like a tuple of both values? Or make a separate type to just hold Response and Value? (I thought that was the point of Response<T>, though)

Reply:
Fixed.